node-ui/src/hooks/useInternetComputer.tsx

```diff
@@ -0,0 +1,352 @@
+import { useCallback, useState } from 'react';
+import { randomBytes } from 'crypto';
+import { getOrCreateKeypair } from '../auth/ed25519';
+
+import { DelegationIdentity } from '@dfinity/identity';
+
+import apiClient from '../api';
+import {
+  InternetComputerSignatureMessageMetadata,
+  NodeChallenge,
+  Payload,
+  SignatureMessage,
+  SignatureMessageMetadata,
+  SignData,
+  WalletSignatureData,
+  LoginRequest,
+  LoginResponse,
+  WalletMetadata,
+  WalletType,
+} from '../api/dataSource/NodeDataSource';
+import { ResponseData } from '../api/response';
+import { useNavigate } from 'react-router-dom';
+import { setStorageNodeAuthorized } from '../auth/storage';
+import { useServerDown } from '../context/ServerDownContext';
+import translations from '../constants/en.global.json';
+
+interface RequestNodeDataProps {
+  setErrorMessage: (msg: string) => void;
+}
+
+interface useInternetComputerReturn {
+  ready: boolean;
+  walletSignatureData: WalletSignatureData | null;
+  signData: SignData | null;
+  signMessageAndLogin: (
+    isLogin: boolean,
+    setErrorMessage: (msg: string) => void,
+  ) => void;
+  logout: (setErrorMessage: (msg: string) => void) => void;
+  requestNodeData: ({ setErrorMessage }: RequestNodeDataProps) => void;
+  changeNetwork: (
+    networkId: NetworkId,
+    setErrorMessage: (msg: string) => void,
+  ) => void;
+}
+
+const t = translations.useInternetComputer;
+const possibleNetworks = {
+  production: {
+    url: 'https://identity.ic0.app',
+    cannisterId: 'rdmx6-jaaaa-aaaaa-aaadq-cai',
+  },
+  staging: {
+    url: 'https://beta.identity.ic0.app/',
+    cannisterId: 'fgte5-ciaaa-aaaad-aaatq-cai',
+  },
+};
+
+export type NetworkId = keyof typeof possibleNetworks;
```

Comment:
Should this be the reusable type?

Reply:
No, this type (NetworkId) is only relevant in the context of the Internet Computer (ICP) environments (e.g., production and staging) and is not designed to be transferable to other networks.